### DISCUSSION TITLE
Netherlands: Coolblue Energie (2024)

### DISCUSSION THREAD

**Discussion by rdegraafwhizzkit** [JaccoR/hass-entso-e#162] [Price Modifyer Templates]
Zet VAT tariff op 0 want de genoemde tarieven in het Coolblue Energie contract zijn inclusief BTW, maar de current_price zonder. De berekening wordt volledig gedaan in het template:
```
{% set s = {
    "inkoopvergoeding": 0.020000,
    "energiebelasting": 0.131648,
    "opslag_duurzame_energie": 0.000000,
    "btw": 1.21
  }
%}
{{current_price * s.btw + s.inkoopvergoeding + s.energiebelasting + s.opslag_duurzame_energie | float}}
```

![coolblue_energie](https://github.com/JaccoR/hass-entso-e/assets/13347371/33c5dc9e-304c-4517-8542-83570b1d0da8)

Als ik deze berekening vergelijk met de Coolblue App, dan klopt het resultaat dat uit bovenstaand template komt.

**Comment by rdegraafwhizzkit** [comment]
Test script:
```
from entsoe import EntsoePandasClient
import pandas as pd


class CoolblueEnergie:
    inkoopvergoeding = 0.020000
    energiebelasting = 0.131648
    opslag_duurzame_energie = 0.000000
    btw = 1.21

    @staticmethod
    def prijs(mwh_prijs: float) -> float:
        return round(
            (mwh_prijs / 1000.0) * CoolblueEnergie.btw +
            CoolblueEnergie.inkoopvergoeding +
            CoolblueEnergie.energiebelasting +
            CoolblueEnergie.opslag_duurzame_energie,
            2
        )


client = EntsoePandasClient(api_key='***')

df = client.query_day_ahead_prices(
    country_code='NL',
    start=pd.Timestamp(year=2024, month=7, day=2, tz='Europe/Amsterdam'),
    end=pd.Timestamp(year=2024, month=7, day=3, tz='Europe/Amsterdam')
)

for index, value in df.items():
    print(f"Uur : {index}, Prijs : {CoolblueEnergie.prijs(value)}")
````
levert netjes
![python](https://github.com/JaccoR/hass-entso-e/assets/13347371/5612df0c-6c77-4986-a7f6-2d6807a9a0b8)
zoals in de app:
![Screenshot_2024-07-02-23-54-22-221-edit_eu coolblue shop](https://github.com/JaccoR/hass-entso-e/assets/13347371/53eaff1c-6ffd-483a-a604-61de6f7f1539)

**Comment by Fastjur** [comment]
Hey hoi, kun jij me helpen met beredeneren waarom dit niet goed gaat voor mijn huidige contract?
Dit zijn mijn prijzen:
![image](https://github.com/user-attachments/assets/c3dd5753-2b57-4180-99e6-ec8210bbbafd)

Dit is wat ik als price modifier template op geef:
```jinja2
{% set s = {
  "inkoopvergoeding": 0.020000,
  "energiebelasting": 0.122863,
  "opslag_duurzame_energie": 0.000000,
  "btw": 1.21
  }
%}
{{(current_price * s.btw + s.inkoopvergoeding + s.energiebelasting + s.opslag_duurzame_energie) | float | round(2)}}
```

Dit is wat mijn home assistant berekent (0.35 cent vanaf 07:00)
<img width="522" alt="Screenshot 2025-03-04 at 13 41 32" src="https://github.com/user-attachments/assets/e653ea2a-e356-41ff-95a0-8575764ec672" />

Maar dit is wat het echt is volgens de coolblue website:
![image](https://github.com/user-attachments/assets/9ade18d9-9e31-4f80-b29e-ab791b88ec9f)

Hij zit er vaak net een cent onder, dus ik snap m niet helemaal!

**Reply by rdegraafwhizzkit** [reply]
Hmmm ja interessant. Ik zie het bij mij inderdaad ook. Eerlijk gezegd heb ik geen idee. Misschien (nu) toch ergens een verrekening met de opslag duurzame energie in de Coolblue app. Het was altijd 1:1 goed...

**Reply by Fastjur** [reply]
Hmm gek. Oh well, het is close enough. Als ik een oplossing vind laat ik t je weten.

**Reply by DrDirtyDevil** [reply]
heeft iemand hier nog iets op gevonden ?

**Reply by Fastjur** [reply]
Ik heb lekker het 1ct verschil voor lief genomen haha.

**Comment by erry2907** [comment]
Bedankt voor jullie instructies,

Ik ben op dit moment een beetje in gevecht met Coolblue, er klopt geen klap van de weergave in de coolblue app of de online omgeving 
ze halen namelijk van de terug geleverde energie van de opgenomen energie af, maar de terug geleverde energie is weergegeven als een negatief getal, dus - en - word dus + .

Daarboven op komt dat de energie tarieven afgerond worden naar boven, 
Sinds vandaag zijn de prijzen in de app bijna 15 cent lager zijn dan ze volgens de site https://www.stroomperuur.nl/ horen te zijn, 

Na wat bellen en heen en weer mailen, kreeg ik het volgende antwoord:

"
Hoi Erwin,
 
Bedankt voor je belletje en je scherpe vragen over de verschillen tussen het stroomverbruik en de kostenweergave in de app, op de website en in externe tools zoals Home Assistant en stroomperuur.nl.
 
Graag geef ik je hier wat meer duidelijkheid over:
Facturen zijn leidend: De facturen worden niet opgesteld op basis van de gegevens in Mijn Coolblue, maar op basis van de daadwerkelijke meterstanden en tarieven. De bedragen op je facturen zijn dus altijd leidend en zouden moeten kloppen.
 
Bekende weergaveproblemen: We zijn op de hoogte van problemen met de weergave van energiekosten en dynamische tarieven in Mijn Coolblue. Hier wordt aan gewerkt. Voor de meest betrouwbare informatie over je kosten en verbruik kun je het beste je facturen raadplegen.
 
Afronding van tarieven: In de app ronden we de tarieven af op 2 decimalen. Dit doen we om het overzichtelijk te houden. Eventuele verschillen na de tweede decimaal zijn minimaal en zorgen hooguit voor een prijsverschil van een paar cent per maand.
 
Dynamische tarieven: Bij de dynamische tarieven zit er momenteel steeds dezelfde afrondingsfout in de weergave, waardoor er een verschil van 1 cent kan ontstaan. Dit heeft geen invloed op de daadwerkelijke facturatie.
 
Energiebelasting: De energiebelasting hoort niet per uur te wijzigen. Mogelijk zorgt dit voor het prijsverschil van 1 cent dat je ziet. Ook hier geldt: de bedragen op je factuur zijn juist.
 
Home Assistant en externe websites: Je hoeft geen afwijkende waardes in te vullen in Home Assistant of andere externe tools. De fout ligt in de weergave van Mijn Coolblue, niet in je eigen instellingen.
 
Hopelijk geeft dit wat meer duidelijkheid! Mocht je nog vragen hebben of ergens over twijfelen, laat het gerust weten. We helpen je graag verder.
Met vriendelijke groet,

C####
Klantenservicemedewerker

"

Dus conclusie is denk ik komende tijd even scherp de facturen in de gaten houden.

**Reply by Fastjur** [reply]
Ik zal toch eens kijken of ik intern iemand bij Coolblue kan vinden die me er wat meer verduidelijking op kan geven, ik zie in iedergeval net wat andere waardes in mijn eigen berekening tov stroomperuur.nl, ik bereken het zelf ongeveer een halve cent lager dan stroomperuur.nl.

Het is lastig te zeggen met de coolblue app er naast, aangezien die prijzen zijn afgerond op centen idd.
